file/builder.go

```diff
@@ -768,9 +786,101 @@ func (b *stateBuilder) ingestRoute(r FRoute) error {
 		p.Route = &kong.Route{ID: kong.String(*r.ID)}
 		plugins = append(plugins, *p)
 	}
+	plugins, err = b.ingestPluginDefaults(plugins)
+	if err != nil {
+		return err
+	}
 	return b.ingestPlugins(plugins)
 }
 
+func ingestDefaultConfigFields(p *FPlugin, config interface{}) {
+	for _, field := range config.([]interface{}) {
+		for key, value := range field.(map[string]interface{}) {
+			if p.Config[key] != nil {
+				continue
+			}
+			if defaultValue, ok := value.(map[string]interface{})["default"]; ok {
+				p.Config[key] = defaultValue
+			} else if fields, ok := value.(map[string]interface{})["fields"]; ok {
+				innerField := map[string]interface{}{}
+				for _, field := range fields.([]interface{}) {
+					for fieldKey, value := range field.(map[string]interface{}) {
+						if defaultValue, ok := value.(map[string]interface{})["default"]; ok {
+							innerField[fieldKey] = defaultValue
+						} else {
+							innerField[fieldKey] = nil
+						}
+					}
+				}
+				p.Config[key] = innerField
+			} else {
+				p.Config[key] = nil
+			}
+		}
+	}
+}
+
+func ingestDefaultProtocols(p *FPlugin, protocols interface{}) {
+	if defaultValue, ok := protocols.(map[string]interface{})["default"]; ok {
+		for _, protocol := range defaultValue.([]interface{}) {
+			if protocolStr, ok := protocol.(string); ok {
+				p.Protocols = append(p.Protocols, &protocolStr)
+			}
+		}
+	}
+}
+
+func (b *stateBuilder) getDefaults(entity, entityID string) (map[string]interface{}, error) {
```

Comment:
```suggestion
func (b *stateBuilder) getDefaults(pluginName string) (map[string]interface{}, error) {
```

Simplify this function for plugin defaults.